dotnet/src/Planners/Planners.Core/Extensions/ReadOnlyFunctionCollectionPlannerExtensions.cs

```diff
@@ -109,20 +147,18 @@ public static class SKContextSequentialPlannerExtensions
     }
 
     private static async Task<IEnumerable<FunctionView>> GetRelevantFunctionsAsync(
-        SKContext context,
         IEnumerable<FunctionView> availableFunctions,
         IAsyncEnumerable<MemoryQueryResult> memories,
+        ILogger? logger = null,
```

Comment:
nit: to remove the null check you could use a NullLogger
```suggestion
        ILogger? logger = NullLogger.Instance,
```